lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
+    # Translate
+    #   VALUE args[6], failed, a1, a2, a3, a4, a5, a6;
+    #  into
+    #   VALUE failed, a1, a2, a3, a4, a5, a6; VALUE *args = truffle_managed_malloc(6 * sizeof(VALUE));
+
+    simple = []
+    arrays = []
+
+    line = $1
+
+    $2.split(', ').each do |local|
```

Comment:
But I'm not just dividing the array into two - I'm converting the values as well, so I'd have to partition and then map the two arrays returned, or convert the strings in place for one array and then return false for values to go into it.